### PR TITLE
[FIX] web: Fixed config layout wizard irresponsive logo

### PR DESCRIPTION
**Pull request by hsal-odoo**
This commit fixes invoice config layout not updating logo live the same way it updates other parts of the invoice (e.g., font color).
This was fixed by reverting parts of the commit https://github.com/odoo/odoo/commit/45d924b7c8e3a2a056a940130591c6bdc0263373.
Moreover, it fixes the invoice preview styles glitch happening at the wizard closing.
This was fixed by avoiding recomputing the preview at wizard closing.

Before this commit, updating the logo did not update the document preview correctly.
Also, there was a glitch when saving the wizard that was caused by the "_compute_preview" being retrigerred.

task-3845046

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/models/base_document_layout.py [hunk 1]
```diff
@@ -120,6 +120,11 @@ def _compute_logo_colors(self):
     @api.depends('report_layout_id', 'logo', 'font', 'primary_color', 'secondary_color', 'report_header', 'report_footer', 'layout_background', 'layout_background_image', 'company_details')
     def _compute_preview(self):
         """ compute a qweb based preview to display on the wizard """
+
+        if self.env['base.document.layout'].browse(self.id):
```
Comment (malb-odoo) [inline addons/web/models/base_document_layout.py:124]: Can you put a comment to explain this, seems a bit magical or hacky without a proper comment to explain why we need to do that

File: addons/web/models/base_document_layout.py [hunk 2]
```diff
@@ -136,7 +141,12 @@ def _compute_preview(self):
 
     def _get_render_information(self, styles):
         self.ensure_one()
-        wizard_with_logo = self.with_context(bin_size=False)
+        # guarantees that bin_size is always set to False,
+        # so the logo always contains the bin data instead of the binary size
+        if self.env.context.get('bin_size'):
+            wizard_with_logo = self.with_context(bin_size=False)
+        else:
+            wizard_with_logo = self
```
Comment (malb-odoo) [inline addons/web/models/base_document_layout.py:149]: All this dif is a revert from what was done here https://github.com/odoo/odoo/commit/45d924b7c8e3a2a056a940130591c6bdc0263373, Can you specify in you commit message that.

**Review by malb-odoo — CHANGES_REQUESTED** [review]
The commit message is a bit weird i have the feeling that you are repeating yourself. I would write something like: 
Before this commit, updating the logo did not update the preview correctly, to solve that we [insert the explanation of the fix]. More over there was a glitch when saving the wizard that was caused by the fact that the compute "_compute_preview" is retriggered ....

**Review by malb-odoo — APPROVED** [review]
LGTM even if the "if" statement is a bit hacky i don't think we can do another way, good job !

**Comment by juliusc2066** [comment]
Hello @hsal-odoo, this fix is a bit fragile since it breaks the preview in case of a save like here https://www.odoo.com/odoo/project/133/tasks/4137809. Could you investigate if the mentioned glitch can be fixed in another way?
Thanks :)

**Pull request closed without merging**